src/main/java/at/hannibal2/skyhanni/features/mining/FlowstateHelper.kt

```diff
@@ -0,0 +1,180 @@
+package at.hannibal2.skyhanni.features.mining
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.api.event.HandleEvent
+import at.hannibal2.skyhanni.data.MiningAPI
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.IslandChangeEvent
+import at.hannibal2.skyhanni.events.ItemInHandChangeEvent
+import at.hannibal2.skyhanni.events.LorenzTickEvent
+import at.hannibal2.skyhanni.events.mining.OreMinedEvent
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.RenderUtils.renderRenderables
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.SimpleTimeMark.Companion.fromNow
+import at.hannibal2.skyhanni.utils.SkyBlockItemModifierUtils.getEnchantments
+import at.hannibal2.skyhanni.utils.TimeUnit
+import at.hannibal2.skyhanni.utils.TimeUtils.format
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+
+@SkyHanniModule
+object FlowstateHelper {
+    private val config get() = SkyHanniMod.feature.mining.flowstateHelper
+    private var streakEndTimer = SimpleTimeMark.farPast()
+    private var blockBreakStreak = 0
+
+    private var display: List<Renderable>? = null
+    private var displayDirty = false
+    private var displayHibernating = true
+    private var timeSinceHibernation = SimpleTimeMark.farPast()
+
+    private var flowstateCache: Int? = null
+
+    private val pickobulusPattern by RepoPattern.pattern(
+        "mining.pickobulus.blockdestroy",
+        "§7Your §r§aPickobulus §r§7destroyed §r§e(?<amount>\\d+) §r§7blocks!"
+    )
+
+    enum class GUIElements(val label: String, var renderable: Renderable = Renderable.string("")) {
```

Comment:
This enum class name should specify that it is for flowstate helper

Reply:
well, https://github.com/hannibal002/SkyHanni/pull/2561#discussion_r1770763611

Reply:
> well, [#2561 (comment)](https://github.com/hannibal002/SkyHanni/pull/2561#discussion_r1770763611)

😭 im not a fan of typing GUIElements anywhere else in the mod and getting this random enum. We already have 2 classes named Item that it autocompletes before the minecraft one. But i guess it is up to @hannibal002

Reply:
We have no real consistency over the project for those gui enum names.

I see your argument for avoiding using the same unspecific and abstract enum name for all those cases all over the project. But I don't think this is such a big deal.
When searching for flowstate, you will still find the enum quicky.

My argument against having the enum having the name flowstate is that the access (in the config class) looks like "FlowstateFeatures.FlowstateGUI".

This is just a visual problem, and there are different solutions to avoid it.
E.g. moving the "flowstate display element enum" into a separate class (not separate file) or using element imports in the config file, or using a abstract class name for all enums that have the same purpose (e.g. GUIElement) or something.

In the end, I'm not advocating for something specific here. If one of you thinks a different name, or generally a different naming convention for those config enum display classes is better, please use it.

I'm not overruling anything that makes sense and my suggestion earlier was not the only solution that makes sense.